build_tools/bazel_to_cmake/bazel_to_cmake_targets.py

```diff
@@ -34,10 +32,17 @@
     "@llvm-project//llvm:X86AsmParser": ["IREELLVMCPUTargetDeps"],
     "@llvm-project//llvm:X86CodeGen": ["IREELLVMCPUTargetDeps"],
 
+    # LLD
+    "@llvm-project//lld": ["${IREE_LLD_TARGET}"],
+    "@llvm-project//lld:COFF": ["lldCOFF"],
+    "@llvm-project//lld:Common": ["lldCommon"],
+    "@llvm-project//lld:ELF": ["lldELF"],
+    "@llvm-project//lld:MachO": ["lldMachO"],
+    "@llvm-project//lld:Wasm": ["lldWasm"],
```

Comment:
These look pretty standardized. We should handle it as a function with only lld special cased here (I'm still a bit suspicious of the lld special case also).